Sporadic error flushing bucket in integration test

Another failure:

http://uberjenkins.sc.couchbase.com/view/Build/job/build-sync-gateway-w-coverage/3/console

```
--- FAIL: TestResponseEncoding (1.38s)
panic: Error trying to empty bucket.  Spec: {Server:http://integration-cbs.sc.couchbase.com:8091 PoolName: BucketName:test_data_bucket FeedType: Auth:{Username:test_data_bucket Password:password BucketName:test_data_bucket} CouchbaseDriver:GoCBCustomSGTranscoder MaxNumRetries:0 InitialRetrySleepTimeMS:0 UseXattrs:false ViewQueryTimeoutSecs:<nil>}.  err: {"_":"Flush failed with unexpected error. Check server logs for details."} [recovered]
	panic: Error trying to empty bucket.  Spec: {Server:http://integration-cbs.sc.couchbase.com:8091 PoolName: BucketName:test_data_bucket FeedType: Auth:{Username:test_data_bucket Password:password BucketName:test_data_bucket} CouchbaseDriver:GoCBCustomSGTranscoder MaxNumRetries:0 InitialRetrySleepTimeMS:0 UseXattrs:false ViewQueryTimeoutSecs:<nil>}.  err: {"_":"Flush failed with unexpected error. Check server logs for details."}

goroutine 24622 [running]:
panic(0xb8a000, 0xc4252fbd30)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc4242563c0)
	/usr/local/go/src/testing/testing.go:579 +0x25d
panic(0xb8a000, 0xc4252fbd30)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/couchbase/sync_gateway/base.GetBucketOrPanicCommon(0x0, 0xb, 0xffffffffffffffff)
	github.com/couchbase/sync_gateway/base/_obj/util_testing.go:127 +0x738
github.com/couchbase/sync_gateway/base.GetBucketOrPanic(0xc4267ca400, 0xf0)
	github.com/couchbase/sync_gateway/base/_obj/util_testing.go:40 +0x38
github.com/couchbase/sync_gateway/rest.(*RestTester).Bucket(0xc424be4000, 0x1, 0xc420ebb330)
	github.com/couchbase/sync_gateway/rest/_test/_obj_test/utilities_testing.go:33 +0x98
github.com/couchbase/sync_gateway/rest.(*RestTester).ServerContext(0xc424be4000, 0xc420ebb330)
	github.com/couchbase/sync_gateway/rest/_test/_obj_test/utilities_testing.go:130 +0x39
github.com/couchbase/sync_gateway/rest.(*RestTester).Send(0xc424be4000, 0xc4267ca4b0, 0xca855d)
	github.com/couchbase/sync_gateway/rest/_test/_obj_test/utilities_testing.go:222 +0x15b
github.com/couchbase/sync_gateway/rest.(*RestTester).SendRequest(0xc424be4000, 0xc9d533, 0x3, 0xca855d, 0xf, 0xc4201ba400, 0x180c, 0xc428915d47)
	github.com/couchbase/sync_gateway/rest/_test/_obj_test/utilities_testing.go:193 +0x87
github.com/couchbase/sync_gateway/rest.TestResponseEncoding(0xc4242563c0)
	/var/jenkins/workspace/build-sync-gateway-w-coverage/godeps/src/github.com/couchbase/sync_gateway/rest/api_test.go:1228 +0x1d0
```

--- a/base/util_testing.go
+++ b/base/util_testing.go
@@ -272,9 +272,23 @@ func (tbm *TestBucketManager) BucketItemCount() (itemCount int, err error) {
 
 func (tbm *TestBucketManager) EmptyTestBucket() error {
 
-	if err := tbm.BucketManager.Flush(); err != nil {
+	// Try to Flush the bucket in a retry loop
+	// Ignore sporadic errors like:
+	// Error trying to empty bucket. err: {"_":"Flush failed with unexpected error. Check server logs for details."}
+
+	worker := func() (shouldRetry bool, err error, value interface{}) {
+		err = tbm.BucketManager.Flush()
+		Warn("Error flushing bucket: %v  Will retry.", err)
+		shouldRetry = (err != nil)  // retry (until max attempts) if there was an error
+		return shouldRetry, err, nil
+	}
+	sleeper := CreateDoublingSleeperFunc(20, 100)
+
+	err, _ := RetryLoop("EmptyTestBucket", worker, sleeper)
+	if err != nil {
 		return err
 	}
+
 
 	for {
 

